docs/docs/reference/syntax.md

```diff
@@ -259,8 +259,10 @@ BlockStat         ::=  Import
                     |  Expr1
                     |  EndMarker
 
-ForExpr           ::=  ‘for’ (‘(’ Enumerators ‘)’ | ‘{’ Enumerators ‘}’) {nl} [‘yield’] Expr
-                    |  ‘for’ Enumerators (‘do’ Expr | ‘yield’ Expr)
+ForExpr           ::=  ‘for’ ‘(’ Enumerators0 ‘)’ {nl} [‘do‘ | ‘yield’] Expr
+                    |  ‘for’ ‘{’ Enumerators0 ‘}’ {nl} [‘do‘ | ‘yield’] Expr
+                    |  ‘for’     Enumerators0     {nl} (‘do‘ | ‘yield’) Expr
+Enumerators0      ::=  {nl} Enumerators [semi]
```

Comment:
No `{nl}` needs to be inserted here.

Reply:
Enumerators themselves do not accept new line before them.
> No {nl} needs to be inserted here.

Is it because it's kinda "obvious" that here can go an arbitrary amount of new lines?
For me, it's as obvious as `{nl}` after `)` in `for` / `if` / `while`, still they contain `{nl}` in rules.

Reply:
No, `nl` means an newline token inserted by the lexer. But there is never a newline inserted in front of a `do` or `yield`.